### PR TITLE
CPLAT-10351: v0.2.0

### LINKED ISSUE #31: generateFilePaths excludes files with absolute path
`generateFilePaths` excludes valid file paths because when an absolute path is passed, `filePath` is initialized to a relative path

```dart
@override
Iterable<String> generateFilePaths() sync* {
  final dir = Directory(dirPath);
  for (final fse
      in dir.listSync(followLinks: followLinks, recursive: recursive)) {
    if (fse is! File) {
      continue;
    }
    final filePath = p.relative(fse.absolute.path); // convert to relative path
    if (!pathLooksLikeCode(filePath)) {...}
    ...
}
```

which means that `pathLooksLikeCode` will always return false because the path starts with `.`.

```dart
bool pathLooksLikeCode(String filePath) =>
    !filePath.contains('/.') &&
    !(filePath.startsWith('.') && !filePath.startsWith('./')) &&
    !filePath.startsWith('build/');
```

In my opinion this is unexpected behavior I would propose removing that condition in order to allow for absolute paths. Thoughts? Thanks!

### PR DESCRIPTION
**Pull request by evanweible-wf**
# [CPLAT-10351](https://jira.atl.workiva.net/browse/CPLAT-10351)
Issue Status

## Motivation
Primary motivation is to simplify the writing of codemod scripts by removing `FileQuery` and changing `runInteractiveCodemod()` to accept an `Iterable<File>`. This gives authors much more flexibility in determining which files a codemod runs on.

Fixes #31.

## Changes

See changelog: https://github.com/Workiva/dart_codemod/blob/7c08544c7d5ebf3a0d16ff55fef747762e46f138/CHANGELOG.md

- Remove `FileQuery` (and `createPathFilter()` and `isDartFile()`)
- Change `runInteractiveCodemod()` to accept `Iterable<File>` instead of `FileQuery`
- Recommend usage of `package:glob` in documentation
- Add `filePathsFromGlob()` util as convenience for most common use case – running codemod on files in a project that aren't hidden (e.g. in `.dart_tool`)

Simple codemod consumption now looks like:

```dart
import 'package:codemod/codemod.dart';
import 'package:glob/glob.dart';

void main() {
  runInteractiveCodemod(
    filePathsFromGlob(Glob('**.dart', recursive: true)),
    ExampleSuggestor());
}
```

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/util.dart [hunk 1]
```diff
@@ -78,17 +82,22 @@ void applyPatchesAndSave(SourceFile sourceFile, Iterable<Patch> patches) {
...
   for (final patch in sortedPatches) {
     if (patch.startOffset < lastEdgeOffset) {
       stdout.writeln(
           'A patch that overlaps with a previous patch applied was found. '
           'Do you want to skip this patch, or quit the codemod?\n'
-          'Overlapping patch: ${patch.toString()}\n'
-          'Updated text: ${patch.updatedText}\n'
+          'Previous patch:\n'
+          '  $prev\n'
+          '  Updated text: ${prev.updatedText}\n'
```
Comment (smaifullerton-wk) [inline lib/src/util.dart:97]: won't this throw a NPE the first time?

Comment (smaifullerton-wk) [reply lib/src/util.dart:97]: oh nope, I see the conditional

**Comment by evanweible-wf** [comment]
@Workiva/release-management-p

**Pull request merged**